<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Case study - single cell SingleCellExperiment â€¢ tidytranscriptomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Case study - single cell SingleCellExperiment">
<meta property="og:description" content="tidytranscriptomics">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tidytranscriptomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/comparison_differential_transcription.html">Case study - comparison methods</a>
    </li>
    <li>
      <a href="../articles/deconvolution.html">Case study - deconvolution</a>
    </li>
    <li>
      <a href="../articles/nesting.html">Case study - nesting</a>
    </li>
    <li>
      <a href="../articles/singlecell_tidy_SingleCellExperiment.html">Case study - single cell SingleCellExperiment</a>
    </li>
    <li>
      <a href="../articles/singlecell_tidy_seurat.html">Case study - single cell Seurat</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/stemangiola/tidytranscriptomics/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="singlecell_tidy_SingleCellExperiment_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Case study - single cell SingleCellExperiment</h1>
                        <h4 class="author">Maria Doyle, Peter MacCallum Cancer Centre<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
                        <h4 class="author">Stefano Mangiola, Walter and Eliza Hall Institute<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/stemangiola/tidytranscriptomics/blob/master/vignettes/singlecell_tidy_SingleCellExperiment.Rmd"><code>vignettes/singlecell_tidy_SingleCellExperiment.Rmd</code></a></small>
      <div class="hidden name"><code>singlecell_tidy_SingleCellExperiment.Rmd</code></div>

    </div>

    
    
<p>Recommended Background Reading <a href="https://melbournebioinformatics.github.io/r-intro-biologists/intro_r_biologists.html">Introduction to R for Biologists</a></p>
<p>SingleCellExperiment is a very popular container for single cell RNA sequencing data <span class="citation">[@sce]</span>.</p>
<p>tidySCE provides a bridge between the SingleCellExperiment single-cell package <span class="citation">[@sce]</span> and the tidyverse <span class="citation">[@wickham2019welcome]</span>. It enables the display of the SingleCellExperiment object as a tidyverse tibble, and provides SingleCellExperiment-compatible <em>dplyr</em>, <em>tidyr</em>, <em>ggplot</em> and <em>plotly</em> functions.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load additional libraries</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biocasia2020tidytranscriptomics</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://purrr.tidyverse.org">purrr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://stringr.tidyverse.org">stringr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scater</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">igraph</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SingleR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://www.r-project.org">tidyHeatmap</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">tidySCE</span><span class="op">)</span></code></pre></div>
<div id="create-tidysce" class="section level2">
<h2 class="hasAnchor">
<a href="#create-tidysce" class="anchor"></a>Create <code>tidySCE</code>
</h2>
<p>This is a SingleCellExperiment object but it is evaluated as tibble. So it is fully compatible both with SingleCellExperiment and tidyverse APIs.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_tidy</span> <span class="op">&lt;-</span> <span class="fu">biocasia2020tidytranscriptomics</span><span class="fu">::</span><span class="va">pbmc_small</span> <span class="op">%&gt;%</span> <span class="fu">tidy</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><strong>It looks like a tibble</strong></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_tidy</span></code></pre></div>
<p><strong>But it is a SingleCellExperiment object after all</strong></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assayNames</a></span><span class="op">(</span><span class="va">pbmc_small_tidy</span><span class="op">)</span> <span class="co"># from SummarizedExperiment</span></code></pre></div>
</div>
<div id="polish-the-data" class="section level2">
<h2 class="hasAnchor">
<a href="#polish-the-data" class="anchor"></a>Polish the data</h2>
<p>We can interact with our object as we do with any tibble. In this case we want to polish an annotation column.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_polished</span> <span class="op">&lt;-</span>
    <span class="va">pbmc_small_tidy</span> <span class="op">%&gt;%</span>
    
    <span class="co"># Clean groups</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>groups <span class="op">=</span> <span class="va">groups</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html">str_remove</a></span><span class="op">(</span><span class="st">"^g"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    
    <span class="co"># Extract sample</span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/extract.html">extract</a></span><span class="op">(</span><span class="va">file</span>, <span class="st">"sample"</span>, <span class="st">"../data/sample([a-z0-9]+)/outs.+"</span><span class="op">)</span>
<span class="va">pbmc_small_polished</span></code></pre></div>
</div>
<div id="calculate-the-log-of-the-scaled-counts" class="section level2">
<h2 class="hasAnchor">
<a href="#calculate-the-log-of-the-scaled-counts" class="anchor"></a>Calculate the log of the scaled counts</h2>
<p>We can treat <code>pbmc_small_polished</code> as a <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html"><em>SingleCellExperiment</em> object</a></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">pbmc_small_polished</span>, <span class="st">"counts"</span><span class="op">)</span>                         <span class="co"># from SummarizedExperiment</span>
<span class="va">libsizes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>
<span class="va">size.factors</span> <span class="op">&lt;-</span> <span class="va">libsizes</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">libsizes</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html">logcounts</a></span><span class="op">(</span><span class="va">pbmc_small_polished</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">/</span><span class="va">size.factors</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>  <span class="co"># from SingleCellExperiment</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assayNames</a></span><span class="op">(</span><span class="va">pbmc_small_polished</span><span class="op">)</span>                                        <span class="co"># from SummarizedExperiment</span></code></pre></div>
</div>
<div id="exploratory-analyses" class="section level2">
<h2 class="hasAnchor">
<a href="#exploratory-analyses" class="anchor"></a>Exploratory analyses</h2>
<p>Here we plot abundance of two transcripts for each group.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_polished_abundance</span> <span class="op">=</span> 
  <span class="va">pbmc_small_polished</span> <span class="op">%&gt;%</span>
  
  <span class="co"># Extract abundance</span>
  <span class="fu">join_transcripts</span><span class="op">(</span>transcripts<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"HLA-DRA"</span>, <span class="st">"TCL1A"</span><span class="op">)</span><span class="op">)</span> 
  
<span class="va">pbmc_small_polished_abundance</span></code></pre></div>
<p>We proceed to plot. <em>Note: we donâ€™t need to create many of these temporary variables. They are created here for educational purposes only.</em></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_polished_abundance</span> <span class="op">%&gt;%</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">groups</span>, <span class="va">abundance_counts</span> <span class="op">+</span> <span class="fl">1</span>, fill<span class="op">=</span><span class="va">groups</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span>outlier.shape<span class="op">=</span><span class="cn">NA</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.5</span>, width<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">transcript</span>, scales<span class="op">=</span><span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span>
  <span class="va">custom_theme</span></code></pre></div>
</div>
<div id="reduce-dimensions" class="section level2">
<h2 class="hasAnchor">
<a href="#reduce-dimensions" class="anchor"></a>Reduce dimensions</h2>
<div id="pca" class="section level3">
<h3 class="hasAnchor">
<a href="#pca" class="anchor"></a>PCA</h3>
<p>We proceed with data processing with Bioconductor packages, such as <em>scran</em> <span class="citation">[@lun2016pooling]</span> and <em>scater</em> <span class="citation">[@mccarthy2017scater]</span>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">variable_genes</span> <span class="op">&lt;-</span>
    <span class="va">pbmc_small_polished</span> <span class="op">%&gt;%</span>
    <span class="fu">modelGeneVar</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>                 <span class="co"># from scran</span>
    <span class="fu">getTopHVGs</span><span class="op">(</span>prop<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span>               <span class="co"># from scran</span>
<span class="co"># Perform PCA with scater</span>
<span class="va">pbmc_small_pca</span> <span class="op">&lt;-</span>
    <span class="va">pbmc_small_polished</span> <span class="op">%&gt;%</span>
    <span class="fu">runPCA</span><span class="op">(</span>subset_row<span class="op">=</span><span class="va">variable_genes</span><span class="op">)</span>  <span class="co"># from scater</span>
<span class="va">pbmc_small_pca</span></code></pre></div>
<p>If a tidyverse-compatible package is not included in the tidySCE collection, we can use <code>as_tibble</code> to permanently convert <code>tidySCE</code> into a tibble.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create pairs plot with GGally</span>
<span class="va">pbmc_small_pca</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">GGally</span><span class="fu">::</span><span class="fu"><a href="https://ggobi.github.io/ggally/reference/ggpairs.html">ggpairs</a></span><span class="op">(</span>columns<span class="op">=</span><span class="fl">4</span><span class="op">:</span><span class="fl">8</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>colour<span class="op">=</span><span class="va">groups</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="va">custom_theme</span></code></pre></div>
</div>
<div id="umap" class="section level3">
<h3 class="hasAnchor">
<a href="#umap" class="anchor"></a>UMAP</h3>
<p>Beside PCA which is a linear dimensionality reduction, we can apply neighbour aware methods such as UMAP, to better define locally similar cells. We can calculate the first 3 UMAP dimensions using the scater framework.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_UMAP</span> <span class="op">&lt;-</span>
    <span class="va">pbmc_small_pca</span> <span class="op">%&gt;%</span>
    <span class="fu">runUMAP</span><span class="op">(</span>ncomponents<span class="op">=</span><span class="fl">3</span><span class="op">)</span>           <span class="co"># from scater</span>
<span class="va">pbmc_small_UMAP</span></code></pre></div>
</div>
</div>
<div id="identify-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-clusters" class="anchor"></a>Identify clusters</h2>
<p>We can proceed with cluster identification with <em>scran</em>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_cluster</span> <span class="op">&lt;-</span> <span class="va">pbmc_small_UMAP</span>
<span class="co"># Assign clusters to the 'colLabels' </span>
<span class="co"># of the SingleCellExperiment object</span>
<span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/colLabels.html">colLabels</a></span><span class="op">(</span><span class="va">pbmc_small_cluster</span><span class="op">)</span> <span class="op">&lt;-</span>                                          <span class="co"># from SingleCellExperiment</span>
    <span class="va">pbmc_small_pca</span> <span class="op">%&gt;%</span>
    <span class="fu">buildSNNGraph</span><span class="op">(</span>use.dimred<span class="op">=</span><span class="st">"PCA"</span><span class="op">)</span> <span class="op">%&gt;%</span>                                   <span class="co"># from scran - shared nearest neighbor</span>
    <span class="fu">cluster_walktrap</span><span class="op">(</span><span class="op">)</span> <span class="op">%$%</span>                                                <span class="co"># from igraph</span>
    <span class="va">membership</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="op">)</span>
<span class="co"># Reorder columns</span>
<span class="va">pbmc_small_cluster</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">label</span>, <span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html">everything</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Now we can interrogate the object as if it was a regular tibble data frame.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_cluster</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">groups</span>, <span class="va">label</span><span class="op">)</span></code></pre></div>
<p>And we can plot them using 3D plot using plotly.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_cluster</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://docs.ropensci.org/plotly/reference/plot_ly.html">plot_ly</a></span><span class="op">(</span>
    x <span class="op">=</span> <span class="op">~</span><span class="va">UMAP1</span>,
    y <span class="op">=</span> <span class="op">~</span><span class="va">UMAP2</span>,
    z <span class="op">=</span> <span class="op">~</span><span class="va">UMAP3</span>,
    color <span class="op">=</span> <span class="op">~</span> <span class="va">label</span>,
    colors <span class="op">=</span> <span class="va">friendly_cols</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>
  <span class="op">)</span></code></pre></div>
</div>
<div id="plotting-heatmap" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-heatmap" class="anchor"></a>Plotting heatmap</h2>
<div id="using-scater" class="section level3">
<h3 class="hasAnchor">
<a href="#using-scater" class="anchor"></a>Using scater</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">plotHeatmap</span><span class="op">(</span>                                  <span class="co"># from scater</span>
  <span class="va">pbmc_small_cluster</span>, 
  features<span class="op">=</span><span class="va">variable_genes</span>, 
  columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">pbmc_small_cluster</span><span class="op">$</span><span class="va">label</span><span class="op">)</span>, 
  colour_columns_by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"label"</span><span class="op">)</span>
<span class="op">)</span> </code></pre></div>
</div>
<div id="using-tidyverse" class="section level3">
<h3 class="hasAnchor">
<a href="#using-tidyverse" class="anchor"></a>Using tidyverse</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_cluster</span> <span class="op">%&gt;%</span>
  
  <span class="co"># Get transcript abundance</span>
  <span class="fu">join_transcripts</span><span class="op">(</span>transcripts<span class="op">=</span><span class="va">variable_genes</span><span class="op">)</span> <span class="op">%&gt;%</span>
  
  <span class="co"># Plot heatmap</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidyHeatmap/man/heatmap-methods.html">heatmap</a></span><span class="op">(</span>                                            <span class="co"># from tidyHeatmap</span>
    .row <span class="op">=</span> <span class="va">transcript</span>,
    .column <span class="op">=</span> <span class="va">cell</span>, 
    .value <span class="op">=</span> <span class="va">abundance_logcounts</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>
  
  <span class="co"># Add annotation</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidyHeatmap/man/add_tile-methods.html">add_tile</a></span><span class="op">(</span><span class="va">label</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidyHeatmap/man/add_point-methods.html">add_point</a></span><span class="op">(</span><span class="va">PC1</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="cell-type-classification" class="section level2">
<h2 class="hasAnchor">
<a href="#cell-type-classification" class="anchor"></a>Cell type classification</h2>
<p>We can infer cell type identities using <em>SingleR</em> <span class="citation">[@aran2019reference]</span> and manipulate the output using tidyverse. SingleR <a href="https://bioconductor.org/packages/devel/bioc/vignettes/SingleR/inst/doc/SingleR.html">accepts any log-normalised transcript abundance matrix.</a></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DO NOT EXECUTE - CELLDEX DEPENDENCY NOT IN BIOCONDUCTOR YET</span>
<span class="co"># Get cell type reference data</span>
<span class="va">hpca</span> <span class="op">&lt;-</span> <span class="fu">HumanPrimaryCellAtlasData</span><span class="op">(</span><span class="op">)</span>
<span class="co"># Infer cell identities</span>
<span class="va">cell_type_df</span> <span class="op">&lt;-</span>
    
  <span class="co"># extracting counts from SingleCellExperiment object</span>
  <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">pbmc_small_cluster</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">%&gt;%</span>
    <span class="co"># SingleR</span>
  <span class="fu">SingleR</span><span class="op">(</span>
    ref <span class="op">=</span> <span class="va">hpca</span>,
    labels <span class="op">=</span> <span class="va">hpca</span><span class="op">$</span><span class="va">label.main</span>,
    method <span class="op">=</span> <span class="st">"cluster"</span>,
    clusters <span class="op">=</span> <span class="va">pbmc_small_cluster</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">label</span><span class="op">)</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>
    
    <span class="co"># Formatting results</span>
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span>rownames <span class="op">=</span> <span class="st">"label"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">label</span>, <span class="va">first.labels</span><span class="op">)</span></code></pre></div>
<p>We have pre-calculated the cluster classification.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cell_type_df</span> <span class="op">&lt;-</span> <span class="fu">biocasia2020tidytranscriptomics</span><span class="fu">::</span><span class="va">cell_type_df</span>
<span class="va">cell_type_df</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Join UMAP and cell type info</span>
<span class="va">pbmc_small_cell_type</span> <span class="op">&lt;-</span>
  <span class="va">pbmc_small_cluster</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span>
    <span class="va">cell_type_df</span>, 
    by <span class="op">=</span> <span class="st">"label"</span>
  <span class="op">)</span>
<span class="va">pbmc_small_cell_type</span> </code></pre></div>
<p>We can easily summarise the results. For example, we can see how cell type classification overlaps with cluster classification.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_cell_type</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">label</span>, <span class="va">first.labels</span><span class="op">)</span></code></pre></div>
</div>
<div id="nested-analyses" class="section level2">
<h2 class="hasAnchor">
<a href="#nested-analyses" class="anchor"></a>Nested analyses</h2>
<p>A powerful tool we can use with tidySCE is <code>nest</code>. We can easily perform independent analyses on subsets of the dataset. First we classify cell types in lymphoid and myeloid; then, nest based on the new classification</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_nested</span> <span class="op">&lt;-</span>
  <span class="va">pbmc_small_cell_type</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">first.labels</span> <span class="op">!=</span> <span class="st">"Platelets"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>cell_class <span class="op">=</span> 
           <span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html">if_else</a></span><span class="op">(</span>
             <span class="va">`first.labels`</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Macrophage"</span>, <span class="st">"Monocyte"</span><span class="op">)</span>,
             <span class="st">"myeloid"</span>, 
             <span class="st">"lymphoid"</span>
           <span class="op">)</span>
          <span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/nest-methods.html">nest</a></span><span class="op">(</span>data <span class="op">=</span> <span class="op">-</span><span class="va">cell_class</span><span class="op">)</span>
<span class="va">pbmc_small_nested</span></code></pre></div>
<p>Now we can independently perform analyses for the lymphoid and myeloid cell subsets.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pbmc_small_nested</span> <span class="op">%&gt;%</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>variable_genes <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_chr</a></span><span class="op">(</span>
  <span class="va">data</span>, <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span>
    <span class="fu">modelGeneVar</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">getTopHVGs</span><span class="op">(</span>prop<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span>collapse<span class="op">=</span><span class="st">", "</span><span class="op">)</span>
<span class="op">)</span><span class="op">)</span> </code></pre></div>
</div>
<div id="key-points" class="section level2">
<h2 class="hasAnchor">
<a href="#key-points" class="anchor"></a>Key Points</h2>
<ul>
<li>Some basic steps of a single-cell RNA sequencing analysis are dimensionality reduction, cluster identification and cell type classification</li>
<li>
<code>tidySCE</code> is an invisible layer that operates on a <code>SingleCellExperiment</code> object and enables us to visualise and manipulate data as if it were a tidy data frame.</li>
<li>
<code>tidySCE</code> object is a <code>SingleCellExperiment object</code> so it can be used with any <code>SingleCellExperiment</code> compatible method</li>
</ul>
<pre class="poll poll"><code>Poll: What would be your preferred name for the package currently called tidySCE?</code></pre>
</div>
<div id="contributing" class="section level1">
<h1 class="hasAnchor">
<a href="#contributing" class="anchor"></a>Contributing</h1>
<p>If you want to suggest improvements for this workshop or ask questions, you can do so as described <a href="https://github.com/stemangiola/biocasia2020_tidytranscriptomics/blob/master/CONTRIBUTING.md">here</a>.</p>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>Record package and version information with <code>sessionInfo</code></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:maria.doyle@petermac.org" class="email">maria.doyle@petermac.org</a><a href="#fnref1" class="footnote-back">â†©ï¸Ž</a></p></li>
<li id="fn2"><p><a href="mailto:mangiola.s@wehi.edu.au" class="email">mangiola.s@wehi.edu.au</a><a href="#fnref2" class="footnote-back">â†©ï¸Ž</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Maria Doyle, Stefano Mangiola.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
