<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Producing extensive transcriptomic analysis workflows with tidy principles - 1 hour • tidytranscriptomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Producing extensive transcriptomic analysis workflows with tidy principles - 1 hour">
<meta property="og:description" content="tidytranscriptomics">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tidytranscriptomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/supplementary.html">Supplementary Material</a>
    </li>
    <li>
      <a href="../articles/workshop_1_hour.html">Producing extensive transcriptomic analysis workflows with tidy principles - 1 hour</a>
    </li>
    <li>
      <a href="../articles/workshop_2_hours.html">Producing extensive transcriptomic analysis workflows with tidy principles - 2 hours</a>
    </li>
    <li>
      <a href="../articles/workshop_30_minutes.html">Introduction to tidy transcriptomic analyses - 30 minutes</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/stemangiola/tidytranscriptomics/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workshop_1_hour_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Producing extensive transcriptomic analysis workflows with tidy principles - 1 hour</h1>
                        <h4 class="author">Maria Doyle, Peter MacCallum Cancer Centre<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
                        <h4 class="author">Stefano Mangiola, Walter and Eliza Hall Institute<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/stemangiola/tidytranscriptomics/blob/master/vignettes/workshop_1_hour.Rmd"><code>vignettes/workshop_1_hour.Rmd</code></a></small>
      <div class="hidden name"><code>workshop_1_hour.Rmd</code></div>

    </div>

    
    
<div id="workshop-description" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-description" class="anchor"></a>Workshop Description</h2>
<p>…</p>
<p>The topics presented in this workshop will be</p>
<p>…</p>
<div id="pre-requisites" class="section level3">
<h3 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h3>
<ul>
<li>…</li>
<li>…</li>
</ul>
<p>Recommended Background Reading <a href="https://melbournebioinformatics.github.io/r-intro-biologists/intro_r_biologists.html">Introduction to R for Biologists</a> …</p>
</div>
<div id="workshop-participation" class="section level3">
<h3 class="hasAnchor">
<a href="#workshop-participation" class="anchor"></a>Workshop Participation</h3>
<p>…</p>
</div>
<div id="r-bioconductor-packages-used" class="section level3">
<h3 class="hasAnchor">
<a href="#r-bioconductor-packages-used" class="anchor"></a><em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>tidyverse</li>
<li>tidybulk</li>
<li>tidyHeatmap</li>
<li>edgeR</li>
<li>ggrepel</li>
<li>airway</li>
</ul>
</div>
<div id="time-outline" class="section level3">
<h3 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h3>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>…</td>
<td>…</td>
</tr>
<tr class="even">
<td>…</td>
<td>…</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h3>
<p>…</p>
<div id="learning-goals" class="section level4">
<h4 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h4>
<ul>
<li>To …</li>
<li>To …</li>
</ul>
</div>
<div id="learning-objectives" class="section level4">
<h4 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h4>
<ul>
<li>…</li>
<li>…</li>
<li>…</li>
</ul>
</div>
</div>
<div id="acknowledgements" class="section level3">
<h3 class="hasAnchor">
<a href="#acknowledgements" class="anchor"></a>Acknowledgements</h3>
<p>This material is adapted from an R for RNA sequencing workshop first run <a href="http://combine-australia.github.io/2016-05-11-RNAseq/">here</a>.</p>
<p><img src="../inst/vignettes/tidybulk_logo.png" width="200px"></p>
</div>
</div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>…</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co"># load libraries</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">tibble</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dplyr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">tidyr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">readr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">stringr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">tidyHeatmap</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">tidybulk</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggrepel</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">airway</span>)</pre></body></html></div>
<div id="gate-gene-enrichment" class="section level3">
<h3 class="hasAnchor">
<a href="#gate-gene-enrichment" class="anchor"></a>Gate + gene enrichment</h3>
<div class="sourceCode" id="cb2"><html><body><pre class="r">
<span class="no">cell_type_mu_sigma</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">ancestor</span><span class="kw">==</span><span class="st">"t_CD8"</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html">pivot_wider</a></span>(<span class="kw">names_from</span> <span class="kw">=</span> <span class="no">.which</span>, <span class="kw">values_from</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">.sd</span>)) <span class="kw">%&gt;%</span>
  <span class="kw pkg">tidygate</span><span class="kw ns">::</span><span class="fu">gate</span>(
    <span class="kw">.element</span> <span class="kw">=</span> <span class="no">symbol</span>,
    <span class="kw">.dim1</span> <span class="kw">=</span> <span class="no">sigma_mu</span> ,
    <span class="kw">.dim2</span> <span class="kw">=</span> <span class="no">sigma_sigma</span>,
    <span class="kw">.size</span><span class="kw">=</span><span class="fl">0.01</span>,
    <span class="kw">opacity</span> <span class="kw">=</span> <span class="fl">0.1</span>
  ) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/symbol_to_entrez.html">symbol_to_entrez</a></span>(<span class="kw">.transcript</span> <span class="kw">=</span> <span class="no">symbol</span>, <span class="kw">.sample</span> <span class="kw">=</span> <span class="no">ancestor</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/test_gene_overrepresentation-methods.html">test_gene_overrepresentation</a></span>(<span class="no">ancestor</span>, <span class="no">entrez</span>, <span class="kw">.do_test</span> <span class="kw">=</span> <span class="no">gate</span><span class="kw">==</span><span class="fl">1</span>, <span class="kw">species</span> <span class="kw">=</span> <span class="st">"Homo sapiens"</span>)</pre></body></html></div>
</div>
</div>
<div id="how-to-start-from-tables" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-start-from-tables" class="anchor"></a>How to start from tables</h2>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="co"># create some example tables to use</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">airway</span>)

<span class="co"># counts table</span>
<span class="no">counts</span> <span class="kw">&lt;-</span> <span class="fu">assay</span>(<span class="no">airway</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span>(<span class="kw">rownames</span> <span class="kw">=</span> <span class="st">"geneID"</span>)

<span class="co"># sample information table</span>
<span class="no">sampleinfo</span> <span class="kw">&lt;-</span> <span class="fu">colData</span>(<span class="no">airway</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span>(<span class="kw">rownames</span> <span class="kw">=</span> <span class="st">"sample"</span>)

<span class="co"># data preprocessing</span>

<span class="no">counts_tt</span> <span class="kw">&lt;-</span>
    <span class="co"># convert to tidy format</span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span>(<span class="no">counts</span>, <span class="kw">cols</span> <span class="kw">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">starts_with</a></span>(<span class="st">"SRR"</span>), <span class="kw">names_to</span> <span class="kw">=</span> <span class="st">"sample"</span>, <span class="kw">values_to</span> <span class="kw">=</span> <span class="st">"counts"</span>) <span class="kw">%&gt;%</span>

    <span class="co"># get gene symbols</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/ensembl_to_symbol-methods.html">ensembl_to_symbol</a></span>(<span class="no">geneID</span>) <span class="kw">%&gt;%</span>

    <span class="co"># order the columns for tidybulk</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">sample</span>, <span class="no">geneID</span>, <span class="no">counts</span>, <span class="no">transcript</span>) <span class="kw">%&gt;%</span>

    <span class="co"># add the sample info</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span>(<span class="no">sampleinfo</span>) <span class="kw">%&gt;%</span>

    <span class="co"># shorten sample name</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">sample</span><span class="kw">=</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html">str_remove</a></span>(<span class="no">sample</span>, <span class="st">"SRR1039"</span>)) <span class="kw">%&gt;%</span>

    <span class="co"># convert to tidybulk object</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/tidybulk-methods.html">tidybulk</a></span>(<span class="kw">.sample</span><span class="kw">=</span><span class="no">sample</span>, <span class="kw">.transcript</span><span class="kw">=</span><span class="no">geneID</span>, <span class="kw">.abundance</span><span class="kw">=</span><span class="no">counts</span>)
<span class="co">#&gt; Joining, by = "sample"</span></pre></body></html></div>
</div>
<div id="how-to-count-reads-per-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-count-reads-per-sample" class="anchor"></a>How to count reads per sample</h2>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">counts_tt</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(<span class="no">sample</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span>(<span class="kw">total_reads</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">counts</span>))
<span class="co">#&gt; # A tibble: 8 x 2</span>
<span class="co">#&gt;   sample total_reads</span>
<span class="co">#&gt;   &lt;chr&gt;        &lt;int&gt;</span>
<span class="co">#&gt; 1 508       20637971</span>
<span class="co">#&gt; 2 509       18809481</span>
<span class="co">#&gt; 3 512       25348649</span>
<span class="co">#&gt; 4 513       15163415</span>
<span class="co">#&gt; 5 516       24448408</span>
<span class="co">#&gt; 6 517       30818215</span>
<span class="co">#&gt; 7 520       19126151</span>
<span class="co">#&gt; 8 521       21164133</span></pre></body></html></div>
<p>We can also check how many counts we have for each sample by making a bar plot. This helps us see whether there are any major discrepancies between the samples more easily.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">counts_tt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">sample</span>, <span class="kw">weight</span><span class="kw">=</span><span class="no">counts</span>, <span class="kw">fill</span><span class="kw">=</span><span class="no">sample</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span>() +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()</pre></body></html></div>
<p><img src="workshop_1_hour_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>As we are using ggplot2, we can also easily view by any other variable that’s a column in our dataset, such as cell line, simply by changing <code>fill</code>.</p>
<p>We can colour by dex treatment.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">counts_tt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">sample</span>, <span class="kw">weight</span><span class="kw">=</span><span class="no">counts</span>, <span class="kw">fill</span><span class="kw">=</span><span class="no">dex</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span>() +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()</pre></body></html></div>
<p><img src="workshop_1_hour_files/figure-html/unnamed-chunk-8-1.png" width="700"> We can colour by cell line.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">counts_tt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">sample</span>, <span class="kw">weight</span><span class="kw">=</span><span class="no">counts</span>, <span class="kw">fill</span><span class="kw">=</span><span class="no">cell</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span>() +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()</pre></body></html></div>
<p><img src="workshop_1_hour_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div id="how-to-examine-normalised-counts-with-boxplots" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-examine-normalised-counts-with-boxplots" class="anchor"></a>How to examine normalised counts with boxplots</h2>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="co"># scale counts</span>
<span class="no">counts_scaled</span> <span class="kw">&lt;-</span> <span class="no">counts_tt</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/scale_abundance-methods.html">scale_abundance</a></span>(<span class="kw">factor_of_interest</span> <span class="kw">=</span> <span class="no">dex</span>)

<span class="co"># create box plots</span>
<span class="no">counts_scaled</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(!<span class="no">lowly_abundant</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span>(<span class="kw">cols</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"counts"</span>, <span class="st">"counts_scaled"</span>), <span class="kw">names_to</span> <span class="kw">=</span> <span class="st">"source"</span>, <span class="kw">values_to</span> <span class="kw">=</span> <span class="st">"abundance"</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">sample</span>, <span class="kw">y</span><span class="kw">=</span><span class="no">abundance</span> + <span class="fl">1</span>, <span class="kw">fill</span><span class="kw">=</span><span class="no">dex</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span>() +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_hline</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">yintercept</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span>(<span class="no">abundance</span> + <span class="fl">1</span>)), <span class="kw">colour</span><span class="kw">=</span><span class="st">"red"</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span>(~<span class="no">source</span>) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_log10</a></span>() +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()</pre></body></html></div>
<p><img src="workshop_1_hour_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div id="how-to-create-mds-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-create-mds-plot" class="anchor"></a>How to create MDS plot</h2>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">airway</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/tidybulk-methods.html">tidybulk</a></span>() <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/scale_abundance-methods.html">scale_abundance</a></span>(<span class="kw">factor_of_interest</span><span class="kw">=</span><span class="no">dex</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/reduce_dimensions-methods.html">reduce_dimensions</a></span>(<span class="kw">method</span><span class="kw">=</span><span class="st">"MDS"</span>, <span class="kw">scale</span> <span class="kw">=</span> <span class="no">F</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_sample-methods.html">pivot_sample</a></span>() <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">Dim1</span>, <span class="no">Dim2</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="no">dex</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>()
<span class="co">#&gt; tidybulk says: to access the raw results do `attr(..., "internals")$MDS`</span></pre></body></html></div>
<p><img src="workshop_1_hour_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
</div>
<div id="how-to-create-ma-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-create-ma-plot" class="anchor"></a>How to create MA plot</h2>
<p>MA plots enable us to visualise amount of expression (logCPM) versus logFC. Highly expressed genes are towards the right of the plot. We can also colour significant genes (e.g. genes with FDR &lt; 0.05)</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="co"># perform differential testing</span>
<span class="no">counts_de</span> <span class="kw">&lt;-</span>
    <span class="no">counts_tt</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/test_differential_abundance-methods.html">test_differential_abundance</a></span>(
        <span class="kw">.formula</span> <span class="kw">=</span> ~ <span class="fl">0</span> + <span class="no">dex</span> + <span class="no">cell</span>,
        <span class="kw">.contrasts</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"dextrt - dexuntrt"</span>),
        <span class="kw">omit_contrast_in_colnames</span> <span class="kw">=</span> <span class="fl">TRUE</span>
    )
<span class="co">#&gt; tidybulk says: The design column names are "dextrt, dexuntrt, cellN061011, cellN080611, cellN61311"</span>
<span class="co">#&gt; tidybulk says: to access the raw results (fitted GLM) do `attr(..., "internals")$edgeR`</span>

<span class="co"># maplot, minimal</span>
<span class="no">counts_de</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span>() <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(!<span class="no">lowly_abundant</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">logCPM</span>, <span class="kw">y</span><span class="kw">=</span>-<span class="no">logFC</span>, <span class="kw">colour</span><span class="kw">=</span><span class="no">significant</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>()+
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()</pre></body></html></div>
<p><img src="workshop_1_hour_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>A more informative MA plot, integrating some of the packages in tidyverse.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">counts_de</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span>() <span class="kw">%&gt;%</span>

    <span class="co"># Subset data</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(!<span class="no">lowly_abundant</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">significant</span> <span class="kw">=</span> <span class="no">FDR</span><span class="kw">&lt;</span><span class="fl">0.05</span> <span class="kw">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">logFC</span>) <span class="kw">&gt;=</span><span class="fl">2</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">transcript</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">logFC</span>) <span class="kw">&gt;=</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="no">transcript</span>), <span class="st">""</span>)) <span class="kw">%&gt;%</span>

    <span class="co"># Plot</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">logCPM</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">logFC</span>, <span class="kw">label</span><span class="kw">=</span><span class="no">transcript</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">color</span> <span class="kw">=</span> <span class="no">significant</span>, <span class="kw">size</span> <span class="kw">=</span> <span class="no">significant</span>, <span class="kw">alpha</span><span class="kw">=</span><span class="no">significant</span>)) +
    <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span>() +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span>(<span class="kw">values</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"black"</span>, <span class="st">"#e11f28"</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size_discrete</a></span>(<span class="kw">range</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">2</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()</pre></body></html></div>
<p><img src="workshop_1_hour_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
<div id="how-to-perform-gene-set-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-perform-gene-set-analysis" class="anchor"></a>How to perform gene set analysis</h2>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">clusterProfiler</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">org.Hs.eg.db</span>)

<span class="co"># extract all genes tested for DE</span>
<span class="no">res</span> <span class="kw">&lt;-</span> <span class="no">counts_de_pretty</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/tidybulk/man/pivot_transcript-methods.html">pivot_transcript</a></span>() <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(!<span class="no">lowly_abundant</span>)

<span class="co"># GO terms</span>
<span class="no">egoCC</span> <span class="kw">&lt;-</span> <span class="no">res</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">FDR</span> <span class="kw">&lt;</span> <span class="fl">0.1</span> <span class="kw">&amp;</span> <span class="no">logFC</span> <span class="kw">&gt;</span> <span class="fl">0</span> ) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>( <span class="st">"transcript"</span> ) <span class="kw">%&gt;%</span>
    <span class="fu">enrichGO</span>(
      <span class="kw">OrgDb</span> <span class="kw">=</span> <span class="no">org.Hs.eg.db</span>,
      <span class="kw">keyType</span> <span class="kw">=</span> <span class="st">'SYMBOL'</span>,
      <span class="kw">ont</span> <span class="kw">=</span> <span class="st">"BP"</span>,
      <span class="kw">universe</span> <span class="kw">=</span> (<span class="no">res</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>( <span class="st">"transcript"</span> ) ) )

<span class="fu"><a href="https://rdrr.io/r/graphics/graphics-defunct.html">dotplot</a></span>(<span class="no">egoCC</span>)
<span class="fu">goplot</span>(<span class="no">egoCC</span>)
<span class="fu">emapplot</span>(<span class="no">egoCC</span>)


<span class="co"># MSigDB Hallmark</span>
<span class="no">gmtH</span> <span class="kw">&lt;-</span> <span class="fu">read.gmt</span>( <span class="st">"https://data.broadinstitute.org/gsea-msigdb/msigdb/release/6.2/h.all.v6.2.symbols.gmt"</span> )
<span class="no">enrH</span> <span class="kw">&lt;-</span> <span class="fu">enricher</span>(
   <span class="kw">gene</span> <span class="kw">=</span> ( <span class="no">res</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">FDR</span> <span class="kw">&lt;</span> <span class="fl">0.1</span> <span class="kw">&amp;</span> <span class="no">logFC</span> <span class="kw">&gt;</span> <span class="fl">0</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>( <span class="st">"transcript"</span> ) ),
   <span class="kw">TERM2GENE</span> <span class="kw">=</span> <span class="no">gmtH</span>,
   <span class="kw">universe</span> <span class="kw">=</span> ( <span class="no">res</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>( <span class="st">"transcript"</span> ) ) )

<span class="fu"><a href="https://rdrr.io/r/graphics/graphics-defunct.html">dotplot</a></span>( <span class="no">enrH</span> )
<span class="fu">emapplot</span>(<span class="no">enrH</span>)</pre></body></html></div>
</div>
<div id="key-points" class="section level2">
<h2 class="hasAnchor">
<a href="#key-points" class="anchor"></a>Key Points</h2>
<p>…</p>
</div>
<div id="exercises" class="section level2">
<h2 class="hasAnchor">
<a href="#exercises" class="anchor"></a>Exercises</h2>
<p>…</p>
<p>Questions:<br>
…</p>
</div>
<div id="contributing" class="section level2">
<h2 class="hasAnchor">
<a href="#contributing" class="anchor"></a>Contributing</h2>
<p>If you want to suggest improvements for this workshop or ask questions, you can do so as described <a href="https://github.com/stemangiola/tidytranscriptomics/blob/master/CONTRIBUTING.md">here</a>.</p>
</div>
<div id="reproducibility" class="section level2">
<h2 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h2>
<p>Record package and version information with <code>sessionInfo</code></p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
<span class="co">#&gt; R version 4.0.2 Patched (2020-08-06 r78974)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-openmp/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] airway_1.9.0                SummarizedExperiment_1.19.6</span>
<span class="co">#&gt;  [3] DelayedArray_0.15.7         matrixStats_0.56.0         </span>
<span class="co">#&gt;  [5] Matrix_1.2-18               Biobase_2.49.0             </span>
<span class="co">#&gt;  [7] GenomicRanges_1.41.5        GenomeInfoDb_1.25.10       </span>
<span class="co">#&gt;  [9] IRanges_2.23.10             S4Vectors_0.27.12          </span>
<span class="co">#&gt; [11] BiocGenerics_0.35.4         ggrepel_0.8.2              </span>
<span class="co">#&gt; [13] tidybulk_1.1.5              tidyHeatmap_1.1.4          </span>
<span class="co">#&gt; [15] ggplot2_3.3.2               stringr_1.4.0              </span>
<span class="co">#&gt; [17] readr_1.3.1                 tidyr_1.1.1                </span>
<span class="co">#&gt; [19] dplyr_1.0.1                 tibble_3.0.3               </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] viridis_0.5.1          edgeR_3.31.4           splines_4.0.2         </span>
<span class="co">#&gt;  [4] viridisLite_0.3.0      assertthat_0.2.1       GenomeInfoDbData_1.2.3</span>
<span class="co">#&gt;  [7] yaml_2.2.1             pillar_1.4.6           backports_1.1.8       </span>
<span class="co">#&gt; [10] lattice_0.20-41        limma_3.45.10          glue_1.4.1            </span>
<span class="co">#&gt; [13] digest_0.6.25          RColorBrewer_1.1-2     XVector_0.29.3        </span>
<span class="co">#&gt; [16] colorspace_1.4-1       htmltools_0.5.0        preprocessCore_1.51.0 </span>
<span class="co">#&gt; [19] pkgconfig_2.0.3        GetoptLong_1.0.2       zlibbioc_1.35.0       </span>
<span class="co">#&gt; [22] purrr_0.3.4            scales_1.1.1           farver_2.0.3          </span>
<span class="co">#&gt; [25] generics_0.0.2         ellipsis_0.3.1         withr_2.2.0           </span>
<span class="co">#&gt; [28] cli_2.0.2              magrittr_1.5           crayon_1.3.4          </span>
<span class="co">#&gt; [31] memoise_1.1.0          evaluate_0.14          fs_1.5.0              </span>
<span class="co">#&gt; [34] fansi_0.4.1            MASS_7.3-51.6          tools_4.0.2           </span>
<span class="co">#&gt; [37] hms_0.5.3              GlobalOptions_0.1.2    lifecycle_0.2.0       </span>
<span class="co">#&gt; [40] ComplexHeatmap_2.5.4   locfit_1.5-9.4         munsell_0.5.0         </span>
<span class="co">#&gt; [43] cluster_2.1.0          compiler_4.0.2         pkgdown_1.5.1         </span>
<span class="co">#&gt; [46] rlang_0.4.7            grid_4.0.2             RCurl_1.98-1.2        </span>
<span class="co">#&gt; [49] rjson_0.2.20           circlize_0.4.10        labeling_0.3          </span>
<span class="co">#&gt; [52] bitops_1.0-6           rmarkdown_2.3          gtable_0.3.0          </span>
<span class="co">#&gt; [55] R6_2.4.1               gridExtra_2.3          knitr_1.29            </span>
<span class="co">#&gt; [58] utf8_1.1.4             clue_0.3-57            rprojroot_1.3-2       </span>
<span class="co">#&gt; [61] shape_1.4.4            desc_1.2.0             stringi_1.4.6         </span>
<span class="co">#&gt; [64] Rcpp_1.0.5             vctrs_0.3.2            png_0.1-7             </span>
<span class="co">#&gt; [67] tidyselect_1.1.0       xfun_0.16</span></pre></body></html></div>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:maria.doyle@petermac.org" class="email">maria.doyle@petermac.org</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><a href="mailto:mangiola.s@wehi.edu.au" class="email">mangiola.s@wehi.edu.au</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Maria Doyle.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
