<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Producing extensive transcriptomic analysis workflows with tidy principles - 2 hours • tidytranscriptomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Producing extensive transcriptomic analysis workflows with tidy principles - 2 hours">
<meta property="og:description" content="tidytranscriptomics">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">tidytranscriptomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/supplementary.html">Supplementary Material</a>
    </li>
    <li>
      <a href="../articles/workshop_1_hour.html">Producing extensive transcriptomic analysis workflows with tidy principles - 1 hour</a>
    </li>
    <li>
      <a href="../articles/workshop_2_hours.html">Producing extensive transcriptomic analysis workflows with tidy principles - 2 hours</a>
    </li>
    <li>
      <a href="../articles/workshop_30_minutes.html">Introduction to tidy transcriptomic analyses - 30 minutes</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/stemangiola/tidytranscriptomics/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workshop_2_hours_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Producing extensive transcriptomic analysis workflows with tidy principles - 2 hours</h1>
                        <h4 class="author">Maria Doyle, Peter MacCallum Cancer Centre<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
                        <h4 class="author">Stefano Mangiola, Walter and Eliza Hall Institute<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/stemangiola/tidytranscriptomics/blob/master/vignettes/workshop_2_hours.Rmd"><code>vignettes/workshop_2_hours.Rmd</code></a></small>
      <div class="hidden name"><code>workshop_2_hours.Rmd</code></div>

    </div>

    
    
<div id="workshop-description" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop-description" class="anchor"></a>Workshop Description</h2>
<p>…</p>
<p>The topics presented in this workshop will be</p>
<p>…</p>
<div id="pre-requisites" class="section level3">
<h3 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h3>
<ul>
<li>…</li>
<li>…</li>
</ul>
<p>Recommended Background Reading <a href="https://melbournebioinformatics.github.io/r-intro-biologists/intro_r_biologists.html">Introduction to R for Biologists</a> …</p>
</div>
<div id="workshop-participation" class="section level3">
<h3 class="hasAnchor">
<a href="#workshop-participation" class="anchor"></a>Workshop Participation</h3>
<p>…</p>
</div>
<div id="r-bioconductor-packages-used" class="section level3">
<h3 class="hasAnchor">
<a href="#r-bioconductor-packages-used" class="anchor"></a><em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>tidyverse</li>
<li>tidybulk</li>
<li>tidyHeatmap</li>
<li>edgeR</li>
<li>ggrepel</li>
<li>airway</li>
</ul>
</div>
<div id="time-outline" class="section level3">
<h3 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h3>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>…</td>
<td>…</td>
</tr>
<tr class="even">
<td>…</td>
<td>…</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h3>
<p>…</p>
<div id="learning-goals" class="section level4">
<h4 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h4>
<ul>
<li>To …</li>
<li>To …</li>
</ul>
</div>
<div id="learning-objectives" class="section level4">
<h4 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h4>
<ul>
<li>…</li>
<li>…</li>
<li>…</li>
</ul>
</div>
</div>
<div id="acknowledgements" class="section level3">
<h3 class="hasAnchor">
<a href="#acknowledgements" class="anchor"></a>Acknowledgements</h3>
<p>This material is adapted from an R for RNA sequencing workshop first run <a href="http://combine-australia.github.io/2016-05-11-RNAseq/">here</a>.</p>
<p><img src="../inst/vignettes/tidybulk_logo.png" width="200px"></p>
</div>
</div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>…</p>
<div id="gate-gene-enrichment" class="section level3">
<h3 class="hasAnchor">
<a href="#gate-gene-enrichment" class="anchor"></a>Gate + gene enrichment</h3>
<div class="sourceCode" id="cb1"><html><body><pre class="r">
<span class="co"># Setup data frame</span>
<span class="no">tt</span> <span class="kw">&lt;-</span>

  <span class="co"># Load dataset</span>
  <span class="kw pkg">cellsig</span><span class="kw ns">::</span><span class="no">counts</span> <span class="kw">%&gt;%</span>
  <span class="fu">tidybulk</span>(<span class="no">sample</span>, <span class="no">symbol</span>, <span class="no">count</span>) <span class="kw">%&gt;%</span>

  <span class="co"># Group by level because otherwise samples are duplicated</span>
  <span class="fu">nest</span>(<span class="kw">data</span> <span class="kw">=</span> -<span class="no">level</span>) <span class="kw">%&gt;%</span>

  <span class="co"># Redefine factors inside each level</span>
  <span class="fu">mutate</span>(<span class="kw">data</span> <span class="kw">=</span> <span class="fu">future_map</span>(<span class="no">data</span>, ~ <span class="fu"><a href="https://rdrr.io/r/base/droplevels.html">droplevels</a></span>(<span class="no">.x</span>))) <span class="kw">%&gt;%</span>

  <span class="co"># Fill missing data. There are many genes that</span>
  <span class="co"># are not shared by the majority of samples</span>
  <span class="fu">mutate</span>(<span class="kw">data</span> <span class="kw">=</span> <span class="fu">future_map</span>(<span class="no">data</span>, ~ <span class="fu">fill_missing_abundance</span>(<span class="no">.x</span>, <span class="kw">fill_with</span> <span class="kw">=</span> <span class="fl">0</span>))) <span class="kw">%&gt;%</span>

  <span class="co"># Scale for future PCA plotting</span>
  <span class="fu">mutate</span>(<span class="kw">data</span> <span class="kw">=</span> <span class="fu">future_map</span>(<span class="no">data</span>, ~ <span class="no">.x</span> <span class="kw">%&gt;%</span> <span class="fu">identify_abundant</span>() <span class="kw">%&gt;%</span> <span class="fu">scale_abundance</span>()))

<span class="no">get_constrasts_from_df</span> <span class="kw">=</span> <span class="kw">function</span>(<span class="no">.data</span>){
  <span class="no">.data</span> <span class="kw">%&gt;%</span>

    <span class="fu">distinct</span>(<span class="no">cell_type</span>) <span class="kw">%&gt;%</span>

    <span class="co"># Permute</span>
    <span class="fu">mutate</span>(<span class="kw">cell_type2</span> <span class="kw">=</span> <span class="no">cell_type</span>) <span class="kw">%&gt;%</span>
    <span class="fu">expand</span>(<span class="no">cell_type</span>, <span class="no">cell_type2</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(<span class="no">cell_type</span> <span class="kw">!=</span> <span class="no">cell_type2</span>) <span class="kw">%&gt;%</span>

    <span class="co"># Create contrasts</span>
    <span class="fu">mutate</span>(<span class="kw">contrast</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span>(<span class="st">"cell_type%s - cell_type%s"</span>, <span class="no">cell_type</span>, <span class="no">cell_type2</span>)) <span class="kw">%&gt;%</span>
    <span class="fu">pull</span>(<span class="no">contrast</span>)

}

<span class="no">select_markers_for_each_contrast</span> <span class="kw">=</span> <span class="kw">function</span>(<span class="no">.data</span>){
  <span class="no">.data</span> <span class="kw">%&gt;%</span>

    <span class="co"># Group by contrast. Comparisons both ways.</span>
    <span class="fu">pivot_longer</span>(
      <span class="kw">cols</span> <span class="kw">=</span> <span class="fu">contains</span>(<span class="st">"___"</span>),
      <span class="kw">names_to</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"stats"</span>, <span class="st">"contrast"</span>),
      <span class="kw">values_to</span> <span class="kw">=</span> <span class="st">".value"</span>,
      <span class="kw">names_sep</span><span class="kw">=</span><span class="st">"___"</span>
    ) <span class="kw">%&gt;%</span>

    <span class="co"># Markers selection</span>
    <span class="fu">nest</span>(<span class="kw">stat_df</span> <span class="kw">=</span> -<span class="no">contrast</span>) <span class="kw">%&gt;%</span>

    <span class="co"># Reshape inside each contrast</span>
    <span class="fu">mutate</span>(<span class="kw">stat_df</span> <span class="kw">=</span> <span class="fu">map</span>(<span class="no">stat_df</span>, ~<span class="no">.x</span> <span class="kw">%&gt;%</span> <span class="fu">pivot_wider</span>(<span class="kw">names_from</span> <span class="kw">=</span> <span class="no">stats</span>, <span class="kw">values_from</span> <span class="kw">=</span> <span class="no">.value</span>))) <span class="kw">%&gt;%</span>

    <span class="co"># Rank</span>
    <span class="fu">mutate</span>(<span class="kw">stat_df</span> <span class="kw">=</span> <span class="fu">map</span>(<span class="no">stat_df</span>, ~<span class="no">.x</span> <span class="kw">%&gt;%</span>
                           <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(<span class="no">FDR</span> <span class="kw">&lt;</span> <span class="fl">0.05</span> <span class="kw">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="no">logFC</span>) <span class="kw">&gt;</span> <span class="fl">2</span>) <span class="kw">%&gt;%</span>
                           <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(<span class="no">logCPM</span> <span class="kw">&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(<span class="no">logCPM</span>)) <span class="kw">%&gt;%</span>
                           <span class="fu">arrange</span>(<span class="no">logFC</span> <span class="kw">%&gt;%</span> <span class="fu">desc</span>()) <span class="kw">%&gt;%</span>
                           <span class="fu">slice</span>(<span class="fl">1</span>:<span class="fl">10</span>)
    )) <span class="kw">%&gt;%</span>
    <span class="fu">unnest</span>(<span class="no">stat_df</span>)
}


<span class="no">all_contrasts</span> <span class="kw">&lt;-</span>
  <span class="no">tt</span> <span class="kw">%&gt;%</span>

  <span class="co"># Investigate one level</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(<span class="no">level</span><span class="kw">==</span><span class="fl">1</span>) <span class="kw">%&gt;%</span>

  <span class="co"># Differential transcription</span>
  <span class="fu">mutate</span>(<span class="kw">markers</span> <span class="kw">=</span> <span class="fu">map</span>(
    <span class="no">data</span>,
    ~ <span class="fu">test_differential_abundance</span>(<span class="no">.x</span>,
        ~ <span class="fl">0</span> + <span class="no">cell_type</span>,
        <span class="kw">.contrasts</span> <span class="kw">=</span> <span class="fu">get_constrasts_from_df</span>(<span class="no">.x</span>),
        <span class="kw">action</span><span class="kw">=</span><span class="st">"only"</span>
      )

  )) <span class="kw">%&gt;%</span>

  <span class="co"># Select rank from each contrast</span>
  <span class="fu">mutate</span>(<span class="kw">markers</span> <span class="kw">=</span> <span class="fu">map</span>( <span class="no">markers</span>, ~ <span class="no">.x</span> <span class="kw">%&gt;%</span> <span class="no">select_markers_for_each_contrast</span>)) <span class="kw">%&gt;%</span>

  <span class="co"># Add marker info to original data</span>
  <span class="fu">mutate</span>(<span class="kw">markers</span> <span class="kw">=</span> <span class="fu">map2</span>(<span class="no">markers</span>, <span class="no">data</span>, ~ <span class="fu">left_join</span>(<span class="no">.x</span>, <span class="no">.y</span>))) <span class="kw">%&gt;%</span>
  <span class="fu">select</span>(<span class="no">markers</span>) <span class="kw">%&gt;%</span>
  <span class="fu">unnest</span>(<span class="no">markers</span>) <span class="kw">%&gt;%</span>

  <span class="co"># make contrasts pretty</span>
  <span class="fu">mutate</span>(<span class="kw">contrast_pretty</span> <span class="kw">=</span> <span class="fu">str_replace</span>(<span class="no">contrast</span>, <span class="st">"cell_type"</span>, <span class="st">""</span>) <span class="kw">%&gt;%</span> <span class="fu">str_replace</span>(<span class="st">"cell_type"</span>, <span class="st">""</span>))


<span class="co"># Plot Markers</span>
<span class="no">all_contrasts</span> <span class="kw">%&gt;%</span>
  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="no">cell_type</span>, <span class="no">count_scaled</span> + <span class="fl">1</span>, <span class="kw">color</span><span class="kw">=</span><span class="no">contrast_pretty</span>)) +
  <span class="fu">geom_point</span>(<span class="kw">size</span> <span class="kw">=</span> <span class="fl">0.5</span>) +
  <span class="fu">facet_wrap</span>(~<span class="no">contrast_pretty</span> + <span class="no">symbol</span>, <span class="kw">scale</span><span class="kw">=</span><span class="st">"free_y"</span>) +
  <span class="fu">scale_y_log10</span>() +
  <span class="fu">theme_bw</span>() +
  <span class="fu">theme</span>(
    <span class="kw">text</span> <span class="kw">=</span> <span class="fu">element_text</span>(<span class="kw">size</span><span class="kw">=</span><span class="fl">6</span>),
    <span class="kw">axis.text.x</span> <span class="kw">=</span> <span class="fu">element_text</span>(<span class="kw">angle</span> <span class="kw">=</span> <span class="fl">90</span>, <span class="kw">vjust</span> <span class="kw">=</span> <span class="fl">0.5</span>, <span class="kw">hjust</span><span class="kw">=</span><span class="fl">1</span>)
  )

<span class="co"># Plot PCA</span>
<span class="no">all_contrasts</span> <span class="kw">%&gt;%</span>
  <span class="fu">distinct</span>(<span class="no">sample</span>, <span class="no">symbol</span>, <span class="no">count_scaled</span>,<span class="no">cell_type</span>) <span class="kw">%&gt;%</span>
  <span class="kw pkg">nanny</span><span class="kw ns">::</span><span class="fu">reduce_dimensions</span>(<span class="no">sample</span>, <span class="no">symbol</span>, <span class="no">count_scaled</span>,  <span class="kw">method</span> <span class="kw">=</span> <span class="st">"PCA"</span>, <span class="kw">action</span><span class="kw">=</span><span class="st">"get"</span>, <span class="kw">transform</span> <span class="kw">=</span> <span class="no">log1p</span>) <span class="kw">%&gt;%</span>
  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">PC1</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">PC2</span>, <span class="kw">colour</span> <span class="kw">=</span> <span class="no">cell_type</span>)) +
  <span class="fu">geom_point</span>() +
  <span class="fu">theme_bw</span>()</pre></body></html></div>
</div>
</div>
<div id="key-points" class="section level2">
<h2 class="hasAnchor">
<a href="#key-points" class="anchor"></a>Key Points</h2>
<p>…</p>
</div>
<div id="exercises" class="section level2">
<h2 class="hasAnchor">
<a href="#exercises" class="anchor"></a>Exercises</h2>
<p>…</p>
<p>Questions:<br>
…</p>
</div>
<div id="contributing" class="section level2">
<h2 class="hasAnchor">
<a href="#contributing" class="anchor"></a>Contributing</h2>
<p>If you want to suggest improvements for this workshop or ask questions, you can do so as described <a href="https://github.com/stemangiola/tidytranscriptomics/blob/master/CONTRIBUTING.md">here</a>.</p>
</div>
<div id="reproducibility" class="section level2">
<h2 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h2>
<p>Record package and version information with <code>sessionInfo</code></p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
<span class="co">#&gt; R version 4.0.2 Patched (2020-08-06 r78974)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-openmp/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] rprojroot_1.3-2  digest_0.6.25    crayon_1.3.4     assertthat_0.2.1</span>
<span class="co">#&gt;  [5] MASS_7.3-51.6    R6_2.4.1         backports_1.1.8  magrittr_1.5    </span>
<span class="co">#&gt;  [9] evaluate_0.14    stringi_1.4.6    rlang_0.4.7      fs_1.5.0        </span>
<span class="co">#&gt; [13] rmarkdown_2.3    pkgdown_1.5.1    desc_1.2.0       tools_4.0.2     </span>
<span class="co">#&gt; [17] stringr_1.4.0    yaml_2.2.1       xfun_0.16        compiler_4.0.2  </span>
<span class="co">#&gt; [21] memoise_1.1.0    htmltools_0.5.0  knitr_1.29</span></pre></body></html></div>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:maria.doyle@petermac.org" class="email">maria.doyle@petermac.org</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><a href="mailto:mangiola.s@wehi.edu.au" class="email">mangiola.s@wehi.edu.au</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Maria Doyle.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
